src/Passes/site-packages/TasksCI/cli.py

```diff
@@ -0,0 +1,238 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from .formatting import main as style_check_main
+from .builder import main as builder_main
+from .linting import main as lint_main, clang_tidy_diagnose
+
+import click
+import logging
+import sys
+import os
+import re
+
+# Important directories
+LIB_DIR = os.path.abspath(os.path.dirname((__file__)))
+TEMPLATE_DIR = os.path.join(LIB_DIR, "templates")
+SOURCE_DIR = os.path.abspath(os.path.dirname(os.path.dirname(LIB_DIR)))
+
+# Logging configuration
+logger = logging.getLogger()
+ch = logging.StreamHandler()
+ch.setLevel(logging.DEBUG)
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+ch.setFormatter(formatter)
+logger.addHandler(ch)
+
+# By default we only log errors
+logger.setLevel(logging.ERROR)
+
+
+@click.group()
+@click.option('--loglevel', default="error")
+def cli(loglevel):
+    """
+    Implements the general CLI options such as logging level.
+    """
+
+    # Valid values
+    levels = {
+        "critical": 50,
+        "error": 40,
+        "warning": 30,
+        "info": 20,
+        "debug": 10,
+        "notset": 0
+    }
+
+    # Getting the logging level and updating
+    loglevel = loglevel.lower()
+    if loglevel not in levels:
+        logger.critical("Invalid log level")
+        sys.exit(-1)
+
+    logger.setLevel(levels[loglevel])
+    logger.info("Loglevel set to {}".format(loglevel))
+
+
+@cli.command()
+@click.option('--fix-issues', default=False, is_flag=True)
+def stylecheck(fix_issues):
+    """
+    Command for checking the style and optionally fixing issues. 
+    Note that some issues are not automatically fixed.
+    """
+
+    logger.info("Invoking style checker")
+    style_check_main(fix_issues)
+
+
+@cli.command()
+@click.option("--diagnose", default=False, is_flag=True)
+@click.option('--fix-issues', default=False, is_flag=True)
+@click.option('--force', default=False, is_flag=True)
+def lint(diagnose, fix_issues, force):
+    """
+    Command for linting the code.
+    """
+
+    # Helpful option in order to diagnose Clang tidy.
+    if diagnose:
+        clang_tidy_diagnose()
+
+        # In case we are diagnosing, no run is performed.
+        return
+
+    # Allowing Clang tidy to attempt to fix issues. Generally,
+    # it is discouraged to use this features as it may result in
+    # a catastrophy
+    if fix_issues:
+        if not force:
+            print("""Fixing isssues using Clang Tidy will break your code.
+Make sure that you have committed your changes BEFORE DOING THIS.
+Even so, this feature is experimental and there have been reports of
+clang-tidy modying system libraries - therefore, USE THIS FEATURE AT
+YOUR OWN RISK.
+
+Write 'I understand' to proceed.""")
```

Comment:
This feature is experimental and you will need to reinstall LLVM after running it. If it was a stable feature, I agree it should just ask `y/n`, but at the moment, `clang-tidy` chooses to (that is, is mal-configured) destroy your LLVM headers. The damage is irreversible in the sense that the only solution is to delete you LLVM installation and do a fresh install. Hence, I am reluctant to have a simple yes/no guard as this could give impression that the worst that can happen is that you need to reset your git repo to head and the damage is far beyond what you should possibly expect from a CI script.

If it is a strong preference on your side to that this should just be a yes/no answer, then we can change it.